bears/java/JavaPMDBear.py

```diff
@@ -0,0 +1,98 @@
+        """
+        Checks Java source-code for various problems.
+
+        (``PMD`` is used internally.)
+
+        :param check_best_practices:
+            Activate to check for best practices. See
+            http://pmd.github.io/pmd-5.4.1/pmd-java/rules/index.html#Basic for
+            more information on what this contains.
+        :param check_braces:
+            Check for the right use of braces. See
+            http://pmd.github.io/pmd-5.4.1/pmd-java/rules/index.html#Braces
+            for more information on this matter.
+        :param check_clone_implementation:
+            Check for the right implementation of the ``clone()`` function. See
+            http://pmd.github.io/pmd-5.4.1/pmd-java/rules/index.html#Clone_Implementation
+            for more information on this matter.
+        :param check_code_size:
+            Check for large or complicated code constructs. See
+            http://pmd.github.io/pmd-5.4.1/pmd-java/rules/index.html#Code_Size
+            for more information on this matter.
+        :param check_comments:
+            Check comments for length, content and placement. See
+            http://pmd.github.io/pmd-5.4.1/pmd-java/rules/index.html#Comments
+            for more information on this matter.
+        :param check_controversial:
+            Does various checks that are considered controversial. See
+            http://pmd.github.io/pmd-5.4.1/pmd-java/rules/index.html#Controversial
+            for more information on this matter.
+        :param check_design:
+            Check for optimal code implementations. Alternate approaches
+            are suggested. See
+            http://pmd.github.io/pmd-5.4.1/pmd-java/rules/index.html#Design
+            for more information on this matter.
+        :param check_imports:
+            Check for duplicate and unused imports. See
+            http://pmd.github.io/pmd-5.4.1/pmd-java/rules/index.html#Import_Statements
+            for more information on this matter.
+        :param check_naming:
+            Check the names of identifiers against some rules. See
+            http://pmd.github.io/pmd-5.4.1/pmd-java/rules/index.html#Naming
+            for more information on this matter.
+        :param check_optimizations:
+            Check for best pratices regarding optimization. See
+            http://pmd.github.io/pmd-5.4.1/pmd-java/rules/index.html#Optimization
+            for more information on this matter.
+        :param check_strings:
+            Check for String, StringBuffer and StringBuilder instances. See
+            http://pmd.github.io/pmd-5.4.1/pmd-java/rules/index.html#String_and_StringBuffer
+            for more information on this matter.
+        :param check_unnecessary:
+            Checks for unnecessary code. See
+            http://pmd.github.io/pmd-5.4.1/pmd-java/rules/index.html#Unnecessary
+            for more information on this matter.
+        :param check_unused:
+            Check for unused code. See
```

Comment:
Unused code refers effectively to code that is not used - local variables, private methods, formal parameters, private fields. 
Unnecessary code refers to meaningless code - return statements in a constructor, useless parantheses,  operations on immutable objects like `string.replace(something)` instead of `string = string.replace(something)` :)